src/main/java/frc/robot/subsystem/ElevatorSubsystem.java

```diff
@@ -0,0 +1,366 @@
+
+        elevatorAbsEncoder = MoRotationEncoder.forSparkAbsolute(elevatorA, Units.Rotations);
+        wristAbsEncoder = MoRotationEncoder.forSparkAbsolute(elevatorWrist, Units.Rotations);
+
+        elevatorRelEncoder = MoDistanceEncoder.forSparkRelative(elevatorA, Units.Centimeters);
+        wristRelEncoder = MoRotationEncoder.forSparkRelative(elevatorWrist, Units.Rotations);
+
+        // Setup listeners for encoder scales and absolute zeros. Use notifyImmediately on zero listeners to set the
+        // values now.
+        MoPrefs.elevatorEncoderScale.subscribe(scale -> MoUtils.setupRelativeEncoder(
+                elevatorRelEncoder, elevatorAbsEncoder.getPosition(), MoPrefs.elevatorAbsZero.get(), scale));
+        MoPrefs.elevatorAbsEncoderScale.subscribe(elevatorAbsEncoder::setConversionFactor, true);
+        MoPrefs.wristEncoderScale.subscribe(scale -> MoUtils.setupRelativeEncoder(
+                wristRelEncoder, wristAbsEncoder.getPosition(), MoPrefs.wristAbsZero.get(), scale));
+        MoPrefs.elevatorAbsZero.subscribe(zero -> MoUtils.setupRelativeEncoder(
+                elevatorRelEncoder, elevatorAbsEncoder.getPosition(), zero, MoPrefs.elevatorEncoderScale.get()));
+        MoPrefs.wristAbsZero.subscribe(zero -> MoUtils.setupRelativeEncoder(
+                wristRelEncoder, wristAbsEncoder.getPosition(), zero, MoPrefs.wristEncoderScale.get()));
+
+        MoPrefs.elevatorMaxExtension.subscribe(limit ->
+                elevatorLimitConfig.forwardSoftLimit((float) limit.in(elevatorRelEncoder.getInternalEncoderUnits())));
+        MoPrefs.wristMaxExtension.subscribe(limit ->
+                wristLimitConfig.forwardSoftLimit((float) limit.in(wristRelEncoder.getInternalEncoderUnits())));
+
+        elevatorVelocityPid = new MoSparkMaxElevatorPID(
+                MoSparkMaxPID.Type.VELOCITY, elevatorA, ClosedLoopSlot.kSlot0, elevatorRelEncoder);
+        wristVelocityPid = new MoSparkMaxArmPID(
+                MoSparkMaxPID.Type.VELOCITY,
+                elevatorWrist,
+                ClosedLoopSlot.kSlot0,
+                wristRelEncoder,
+                this::getWristAngleFromHorizontal);
+        elevatorSmartMotionPid = new MoSparkMaxElevatorPID(
+                MoSparkMaxPID.Type.SMARTMOTION, elevatorA, ClosedLoopSlot.kSlot1, elevatorRelEncoder);
+        wristSmartMotionPid = new MoSparkMaxArmPID(
+                MoSparkMaxPID.Type.SMARTMOTION,
+                elevatorWrist,
+                ClosedLoopSlot.kSlot1,
+                wristRelEncoder,
+                this::getWristAngleFromHorizontal);
+
+        elevatorVelTuner = TunerUtils.forMoSparkElevator(elevatorVelocityPid, "Elevator Vel.");
+        wristVelTuner = TunerUtils.forMoSparkArm(wristVelocityPid, "Wrist Vel.");
+        elevatorPosTuner = TunerUtils.forMoSparkElevator(elevatorSmartMotionPid, "Elevator Pos.");
+        wristPosTuner = TunerUtils.forMoSparkArm(wristSmartMotionPid, "Wrist Pos.");
+
```

Comment:
You'll need to call `configureMotors()` here for the configuration to actually happen.